### PR TITLE
Add wcs fitting function in wcs utils 

### PR DESCRIPTION
**Pull request by shannnnnyyy**
This is my first contribution to astropy so if I have done the PR incorrectly let me know. I wrote a function that takes a matched set of pixel positions and sky positions and fits a WCS (linear terms, optional SIP, and keywords based on given projection type and celestial frame). This is similar to the IRAF task CCMAP. Users pass in arrays of X & Y pixel positions, and then a SkyCoord object with the celestial coordinates, as well as the celestial coordinate pair for the projection (or ‘center’), the projection type, and if they would like a SIP fit (and if so, which order for A and B). A WCS object with the best fit parameters as well as other necessary keywords based on the celestial frame and projection type is returned. Optionally, users may pass in an existing WCS object with some keywords already populated and fit or re-fit the linear/polynomial distortion terms with the matched input coordinates. 

I consider this somewhat of a WIP still - I need a few more test cases, especially for with less common projection types, and some feedback on usability and the choices I made for input. Just to note, I use fitters in scipy rather than astropy models and fitters because this was very clunky in my experience to use when fitting a SIP. I do have a notebook with some demonstrative examples that I can link as well. I will submit tests but I’d like feedback on this WIP first. Thanks!

### CODE REVIEW

File: astropy/wcs/utils.py [hunk 1]
```diff
@@ -1,5 +1,3 @@
-# Licensed under a 3-clause BSD style license - see LICENSE.rst
-
```
Comment (bsipocz) [inline astropy/wcs/utils.py]: please don't remove the licence info, neither the empty lines from the docstrings below (sphinx is rather picky for those to be around).

**Comment by nden** [comment]
@shannnnnyyy I think you may have started with an older version of the code. Rebasing may fix things. Let me know if you need help rebasing. Another option is to close this PR and create a new one based on current master.

File: astropy/wcs/utils.py (hunk 1 above)
Comment (astrofrog) [reply astropy/wcs/utils.py]: Related to this, please don't remove all the empty lines from the file

**Comment by astrofrog** [comment]
@shannnnnyyy - just wanted to say thank you for your contribution! I've needed something like this for a long time :) There are some small technical issues with the pull request to fix as we've mentioned above, but I think this will be a very valuable contribution.

**Pull request closed without merging**